### PR TITLE
Python 3.x compatibility changes

### PR DESCRIPTION
**Pull request by lardenoije**
This fixes most issues with Python 3 compatibility (at least it works for me now on Python 3.7). One thing that remains to be fixed is the in_memory option; either it hangs or it just takes a long time, but in either case this also broke the parallel option, causing it to silently fail and write nothing to the output file. So, now that the parallel option no longer uses the in_memory option it works fine. The changes should also make it work again for comb-p.

### CODE REVIEW

**Review by brentp — CHANGES_REQUESTED** [review]
Thanks. This looks good. Can you verify that it works in python2?

File: cruzdb/__init__.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 cruzdb: library for pythonic access to UCSC genome-browser's MySQL database
 """
 from __future__ import print_function
+from past.builtins import long
```
Comment (brentp) [inline cruzdb/__init__.py:5]: This needs a guard for version
```
$ python2.7 -c "from past.builtins import long"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named past.builtins
```

**Review by brentp — COMMENTED** [review]
This looks good. Can you run tests in python2 to make sure everything is OK?

File: cruzdb/__init__.py (hunk 1 above)
Comment (lardenoije) [reply cruzdb/__init__.py]: I didn't get that error with python 2.7.15, but perhaps mine came with the `past.builtins `module pre-installed. Nevertheless, I added a version check before loading the module, as it indeed is not required for python 2.

**Comment by brentp** [comment]
thanks again!

**Pull request merged**